components/DetailView.js

```diff
@@ -35,9 +35,22 @@ class DetailView extends React.Component {
 
   render() {
     const marker = this.props.marker;
-    if (!marker) return <div />;
+    if (!marker) return <div 
+      ref={this.wrapperRef}
+      style={{
+        "opacity" : "0.0"
+      }}/>;
```

Comment:
why do we need these props? the ref is only used to say that if we click on another point in the DetailView, don't close it. opacity 0 also doesn't make sense if there's no children

Reply:
I removed `ref={this.wrapperRef)`, you're right we don't need it.

But we do need opacity to be 0. This is because when someone clicks a marker, we want to the opacity from 0 to 1... so it has to be set to 0 before the user clicks, so that we can use CSS animations to transition it to 1 once clicked.

I found no other way do it that works with CSS Animations, since you can't specify the start or end values when you define the transition.

Reply:
ahhh that's right. dope